UAGimbalAdapter.m

```diff
@@ -75,42 +77,42 @@ - (void)setBluetoothPoweredOffAlertEnabled:(BOOL)bluetoothPoweredOffAlertEnabled
 }
 
 - (void)startWithGimbalAPIKey:(NSString *)gimbalAPIKey {
-    if (self.isStarted) {
-        return;
-    }
-
-    if (gimbalAPIKey.length) {
-        [Gimbal setAPIKey:gimbalAPIKey options:nil];
-    } else if (![GMBLPlaceManager isMonitoring]) {
-        NSLog(@"GMBLPlaceManager is not previously started and API key is not provided. Unable to start Gimbal Adapter.");
-        return;
-    }
-
-    self.placeManager.delegate = self;
-    [GMBLPlaceManager startMonitoring];
+    [Gimbal setAPIKey:gimbalAPIKey options:nil];
+    [Gimbal start];
     self.started = YES;
+    self.placeManager.delegate = self;
+    [self setDeviceAttributes];
     UA_LDEBUG(@"Started Gimbal Adapter. Gimbal application instance identifier: %@", [Gimbal applicationInstanceIdentifier]);
 }
 
-- (void)stop {
-    if (!self.isStarted) {
-        return;
+- (void)setDeviceAttributes {
+    NSMutableDictionary *deviceAttributes = [NSMutableDictionary new];
+    if ([[self.deviceAttributesManager getDeviceAttributes] count] > 0) {
+        [deviceAttributes addEntriesFromDictionary:[self.deviceAttributesManager getDeviceAttributes]];
+    }
+    if ([UAirship namedUser].identifier) {
+        [deviceAttributes setObject:[UAirship namedUser].identifier forKey:@"ua.nameduser.id"];
```

Comment:
you can skip the nil check if you do `[deviceAttributes setValue:[UAirship namedUser].identifier forKey:@"ua.nameduser.id"]`;